plugins/outputs/cloudwatchlogs/cloudwatchlogs.go

```diff
@@ -127,6 +131,20 @@ func (c *CloudWatchLogs) getDest(t Target) *cwDest {
 		},
 	)
 	client.Handlers.Build.PushBackNamed(handlers.NewRequestCompressionHandler([]string{"PutLogEvents"}))
+
+	if ciCompiledRegexp.MatchString(t.Group) {
+		hasCIFlag := false
+		for _, plugin := range agentinfo.OutputPlugins {
+			if plugin == "container_insights" {
+				hasCIFlag = true
+				break
+			}
+		}
+		if !hasCIFlag {
+			agentinfo.OutputPlugins = append(agentinfo.OutputPlugins, "container_insights")
+		}
+		agentinfo.ResetUserAgent()
```

Comment:
I think you want to only reset the UserAgent if you have to set the value, otherwise you're resetting it everytime there is a match.

Reply:
+1. I don't think we want this here. The impetus for adding the `ResetUserAgent()` function is to reset for the tests. It should be used there.

Reply:
I'm going to take this back actually.  Before this change, useragent was a value that stayed static for the entire execution of the agent.  Now we have a situation where sometimes container_insights might be included in the plugins list for user agent and sometimes it might not.  The agent is able to both run container_insights to that log group as well as be configured for other log groups where we do not want that useragent string to have `container_insights`

Reply:
Agreed. @SaxyPandaBear also mentioned `getDest` gets called multiple times. We don't need it to be reset every time there's a match

Reply:
+1 for removing the `agentinfo.ResetUserAgent()` call.
Even if the Agent was configured for uploading multiple log groups and only 1 of which matched the regex, `"^/aws/.*containerinsights/.*/(performance|prometheus)$"` It would still be okay for all uploads from this agent to have a `userAgent` string with `container_insights`.

My interpretation is that the `userAgent` is letting us know which features the customer is using, not which specific feature is associated with this specific upload.

Reply:
As discussed, this has been removed.